### ISSUE TITLE
Upload fails with erroneous "different python packages" error

### ISSUE THREAD

**Issue by itamarst** [PyO3/maturin#518]
Please provide the following information: 

 * Maturin version: 0.10.3
 * Your python version (`python -V`): 3.9.4
 * Your pip version (`pip -V`): 21.1 (latest)
 * The version of the bindings you're using, if any (e.g. `pyo3`, `rust-cpython` or `cffi`): pyo3
 * Does `cargo build` work? N/A

Please list the exact steps required to reproduce your error with all command output and if possible with a repository:

```
$ ls
ahocorasick_rs-0.9.0-cp36-cp36m-macosx_10_7_x86_64.whl
ahocorasick_rs-0.9.0-cp37-cp37m-manylinux1_x86_64.whl
ahocorasick_rs-0.9.0-cp38-none-win_amd64.whl
ahocorasick_rs-0.9.0-cp36-cp36m-manylinux1_x86_64.whl
ahocorasick_rs-0.9.0-cp37-none-win_amd64.whl
ahocorasick_rs-0.9.0-cp39-cp39-macosx_10_7_x86_64.whl
ahocorasick_rs-0.9.0-cp36-none-win_amd64.whl
ahocorasick_rs-0.9.0-cp38-cp38-macosx_10_7_x86_64.whl
ahocorasick_rs-0.9.0-cp39-cp39-manylinux1_x86_64.whl
ahocorasick_rs-0.9.0-cp37-cp37m-macosx_10_7_x86_64.whl
ahocorasick_rs-0.9.0-cp38-cp38-manylinux1_x86_64.whl
ahocorasick_rs-0.9.0-cp39-none-win_amd64.whl
$ maturin upload *.whl
💥 maturin failed
  Caused by: Attempting to upload wheel and/or source distribution files that belong to different python packages.
```

All of those wheels appear to have the same package.

**Comment by itamarst** [comment]
`twine upload *.whl` was perfectly happy to upload everything.

**Comment by itamarst** [comment]
(Other than this minor bug maturin has been _amazing_, by the way, I just wrote the minimal pyproject.toml and it Just Works. Really impressive.)

**Comment by messense** [comment]
The error message comes from here https://github.com/PyO3/maturin/blob/18c134550317ee9e7fd93e063032d50d801dafc6/src/main.rs#L617-L624

Are there any other `.whl` files in the same directory which are not part of `ahocorasick_rs`?

**Comment by itamarst** [comment]
1. There were no other wheels in that directory.
2. They were all built from the same git commit (in different GitHub Actions runners), so metadata ought to be identical but might not be in practice? This is a pretty common use case, "build on Linux+macOS+Windows on different VMs and then upload together".

**Comment by itamarst** [comment]
Perhaps the error should at least say what metadata was different, as a starting point.

**Comment by messense** [comment]
I agree with your points.

I think I have found the cause, the description metadata read from README.md was different because of different newline on different platforms `\n` vs `\r\n`.

**Comment by messense** [comment]
Fixed in #520 

```bash
❯ cargo run upload *.whl
🔐 Using credential in pypirc for upload
🚀 Uploading 12 packages
⛔ Username and/or password are wrong
💥 maturin failed
  Caused by: Username and/or password are wrong
```

**Comment by itamarst** [comment]
Thank you!
